doc/maximal.xml

```diff
@@ -10,102 +11,189 @@
 
 <#GAPDoc Label="MaximalSubsemigroups">
 <ManSection>
-  <Attr Name = "MaximalSubsemigroups" Arg = "S"/>
+  <Attr Name = "MaximalSubsemigroups" Arg = "S" Label="for a finite semigroup"/>
+  <Oper Name = "MaximalSubsemigroups" Arg = "S, opts"
+    Label="for a finite semigroup and a record"/>
   <Returns>
     The maximal subsemigroups of <A>S</A>.
   </Returns>
   <Description>
-    If <A>S</A> is a semigroup, then <C>MaximalSubsemigroups</C> returns
-    a list of the maximal subsemigroups of <A>S</A>.
+    If <A>S</A> is a finite semigroup, then <C>MaximalSubsemigroups</C> returns
+    a list of the non-empty maximal subsemigroups of <A>S</A>.  A <E>maximal
+      subsemigroup</E> of <A>S</A> is a proper subsemigroup of <A>S</A> which is
+    contained in no other proper subsemigroup of <A>S</A>.  The methods used by
+    <C>MaximalSubsemigroups</C> are based on <Cite Key="Graham1968aa"/> and ???.
```

Comment:
@james-d-mitchell Well spotted - I put this in as a placeholder. When our work with Casey is available somewhere, I'll put a reference to it in the documentation along with the GGR reference. For now shall I just remove this?